src/middlewared/middlewared/pytest/unit/utils/test_statx.py

```diff
@@ -0,0 +1,105 @@
+import os
+import pytest
+import stat
+
+from middlewared.plugins.filesystem_ import stat_x as sx
+
+
+BASIC_STAT_ATTRS = [
+    'MODE',
+    'UID',
+    'GID',
+    'ATIME',
+    'MTIME',
+    'CTIME',
+    'DEV',
+    'INO',
+    'SIZE',
+    'BLOCKS',
+    'BLKSIZE',
+    'NLINK',
+]
+
+def timespec_convert(timespec):
+    return timespec.tv_sec + timespec.tv_nsec / 1000000000
+
+
+def do_stat(filename, isdir):
+    if not isdir:
+        with open(filename, "w"):
+            pass
+
+    return (os.stat(filename), sx.statx(filename))
+
+
+def validate_stat(stat_prop, st1, st2):
+    match stat_prop:
+        case 'MODE':
+            assert st1.st_mode == st2.stx_mode
+        case 'UID':
+            assert st1.st_uid == st2.stx_uid
+        case 'GID':
+            assert st1.st_gid == st2.stx_gid
+        case 'ATIME':
+            assert st1.st_atime == timespec_convert(st2.stx_atime)
+        case 'MTIME':
+            assert st1.st_mtime == timespec_convert(st2.stx_mtime)
+        case 'CTIME':
+            assert st1.st_ctime == timespec_convert(st2.stx_ctime)
+        case 'INO':
+            assert st1.st_ino == st2.stx_ino
+        case 'DEV':
+            assert st1.st_dev == os.makedev(st2.stx_dev_major, st2.stx_dev_minor)
+        case 'BLOCKS':
+            assert st1.st_blocks == st2.stx_blocks
+        case 'BLKSIZE':
+            assert st1.st_blksize == st2.stx_blksize
+        case 'NLINK':
+            assert st1.st_nlink == st2.stx_nlink
+        case 'SIZE':
+            assert st1.st_size == st2.stx_size
+        case _:
+            raise ValueError(f'{stat_prop}: unknown stat property')
+
+
+@pytest.mark.parametrize('stat_prop', BASIC_STAT_ATTRS)
+def test__check_statx_vs_stat_file(tmpdir, stat_prop):
+    st1, st2 = do_stat(os.path.join(tmpdir, 'testfile'), False)
+    validate_stat(stat_prop, st1, st2)
+
+
+@pytest.mark.parametrize('stat_prop', BASIC_STAT_ATTRS)
+def test__check_statx_vs_stat(tmpdir, stat_prop):
```

Comment:
Consistent naming convention might be nice.
e.g. `test__check_file_statx_vs_stat` and `test__check_file_statx_vs_stat`